client/src/reducers/posts/reducer.js

```diff
@@ -24,6 +26,114 @@ const postsReducer = (state = innitialState, action) => {
         isLoading: false,
       };
     }
+    case POSTS_ACTIONS.FETCH_REQUESTS_ACTIVE_SUCCESS: {
+      const { payload } = action;
```

Comment:
I am not familiar with Redux, but seems here the logic can be simplified ❔

Reply:
Originally I think it makes sense for feed & profile page to be reusing the same redux action. However, the profile changes into loading a sub-group of posts (active / archived / request / response / all) so reusing the existing state in the store makes it actually much more complicated if I want to cached the data. Thus, I remap the posts to this structure to allow caching to happen. Otherwise, the user will need to wait for a fetch everytime they interact with the component which I think is inefficient. Each query takes about a second, so it's not insignificant.

Reply:
Hmm though maybe there's a way to group all the logic together like @aditilamkhede suggests too. Let me think about it.